src/Actors/Enemy.ts

```diff
@@ -84,15 +96,30 @@ export class Enemy extends Actor {
   }
 
   onCollisionStart(self: Collider, other: Collider, side: Side, contact: CollisionContact): void {
+    if (this.state === "death") {
+      return;
+    }
     if (other.owner instanceof DarkPlayer || other.owner instanceof LightPlayer) {
       (this.scene as GameScene).enemyWaveManager?.enemyPool?.return(this); // Return the enemy to the pool
       this.scene?.remove(this); // Remove the enemy from the scene
       other.owner.currentHP -= 2; // Decrease the player's health by 1
+      const engine = other.owner.scene?.engine;
+      if (engine) {
+        actorFlashWhite(engine, other.owner, 150);
+      }
       this.UISignal.send(["playerDamaged"]);
     }
   }
 
-  onInitialize() {}
+  onInitialize() {
```

Comment:
what you don't know is we're using Object Pooling on the enemies, so we shouldn't use .kill(), we should return to the Object Pool

```ts
 (this.scene as GameScene).enemyWaveManager?.enemyPool?.return(this); // Return the enemy to the pool
      this.scene?.remove(this); // Remove the enemy from the scene
```